DEVELOPER_GUIDE.md

```diff
@@ -374,6 +377,44 @@ The Distribution Download plugin downloads the latest version of OpenSearch by d
 ./gradlew integTest -PcustomDistributionUrl="https://ci.opensearch.org/ci/dbc/bundle-build/1.2.0/1127/linux/x64/dist/opensearch-1.2.0-linux-x64.tar.gz"
 ```
 
+### Creating fat-JAR of a module
+
+A fat-JAR (or an uber-JAR) is the JAR, which contains classes from all the libraries, on which your project depends and, of course, the classes of current project.
+
+There might be cases where the developer would like to add some custom logic to the code of a module (or multiple modules) and generate a fat-JAR that can be directly used by the dependency management tool. For example in [#3665](https://github.com/opensearch-project/OpenSearch/pull/3665) the developer wanted to provide a tentative patch as a fat-JAR to a consumer for changes made in the high level rest client. 
```

Comment:
the developer -> a developer

For example in -> For example, in

rest -> REST